helper/predeployment/predeployment.go

```diff
@@ -0,0 +1,254 @@
+package predeployment
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/0xPolygon/polygon-edge/chain"
+	"github.com/0xPolygon/polygon-edge/contracts/staking"
+	"github.com/0xPolygon/polygon-edge/helper/hex"
+	"github.com/0xPolygon/polygon-edge/state"
+	itrie "github.com/0xPolygon/polygon-edge/state/immutable-trie"
+	"github.com/0xPolygon/polygon-edge/state/runtime"
+	"github.com/0xPolygon/polygon-edge/state/runtime/evm"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/umbracle/ethgo/abi"
+	"io/ioutil"
+	"math"
+	"math/big"
+	"os"
+	"strings"
+)
+
+var (
+	errABINotFound              = errors.New("abi field not found in specified JSON")
+	errBytecodeNotFound         = errors.New("bytecode field not found in specified JSON")
+	errDeployedBytecodeNotFound = errors.New("deployed bytecode field not found in specified JSON")
+)
+
+type contractArtifact struct {
+	ABI              []byte // the ABI of the Smart Contract
+	Bytecode         []byte // the raw bytecode of the Smart Contract
+	DeployedBytecode []byte // the deployed bytecode of the Smart Contract
+}
+
+// generateContractArtifact generates contract artifacts based on the
+// passed in Smart Contract JSON ABI
+func generateContractArtifact(filepath string) (*contractArtifact, error) {
+	var (
+		artifact *contractArtifact
+		err      error
+	)
+
+	// Read from the ABI from the JSON file
+	jsonFile, err := os.Open(filepath)
+	if err != nil {
+		return artifact, err
+	}
+
+	jsonRaw, err := ioutil.ReadAll(jsonFile)
+	if err != nil {
+		return artifact, err
+	}
+
+	// Fill out the fields in the JSON file
+	var jsonResult map[string]interface{}
+	if err = json.Unmarshal(jsonRaw, &jsonResult); err != nil {
+		return artifact, err
+	}
+
+	// Parse the ABI
+	abiRaw, ok := jsonResult["abi"]
+	if !ok {
+		return nil, errABINotFound
+	}
```

Comment:
Just a comment. If you parse json to the struct that the fields typed by json.RawMessage, this field isn't parsed and raw bytes is saved inside.

```
type Hoge struct {
	A json.RawMessage
}

x := Hoge{}
json.Unmarshal([]byte("{\"A\":\"hoge\"}"), &x)

fmt.Println(x.A) // [34 104 111 103 101 34]
```

Reply:
I'm not sure what you meant here?

cc @dbrajovic

Reply:
I said you've parsed json and marshal only ABI again to get ABI in bytes

Reply:
Ah, you meant that we could simply just do:

```go
	// Fill out the fields in the JSON file
	var jsonResult map[string]interface{}
	if err = json.Unmarshal(jsonRaw, &jsonResult); err != nil {
		return nil, err
	}

	// Parse the ABI
	abiBytes, ok := jsonResult[abiValue]
	if !ok {
		return nil, errABINotFound
	}
```

?

Reply:
Since `abiBytes` here would be the same value we get now from `json.Marshal(abiRaw)`?

Reply:
@Kourin1996 ping